spec/support/factories.cr

```diff
@@ -214,6 +301,29 @@ def hg_path(project)
   File.join(tmp_path, project.to_s)
 end
 
+def fossil_commits(project, rev = "trunk")
+  # This is using the workaround code in case the machine running the specs is
+  # using an old Fossil version.  See the #commit_sha1_at method in
+  # src/resolvers/fossil.cr for info.
+  Dir.cd(fossil_path(project)) do
+    retStr = run("fossil timeline #{Process.quote(rev)} -t ci -W 0").strip.split('\n')
+    retLines = retStr.flat_map do |line|
+      /^.+ \[(.+)\].*/.match(line).try &.[1]
+    end
+
+    retLines.reject! &.nil?
+    [/artifact:\s+(.+)/.match(run("fossil whatis #{retLines[0]}")).not_nil!.[1]]
```

Comment:
That mirrors the behavior of the Mercurial code, which this was based on.  That code looks like this:

```crystal
run("hg log --template=#{Process.quote("{node}\n")} -r #{Process.quote(rev)}").strip.split('\n')
```

Let me go ahead and adjust the Fossil-based stuff to not rely on a list since it seems like it's not needed.